### PR TITLE
Add collapser for VTune

### PR DESCRIPTION
**Pull request by jasonrhansen**
Implements #142

### CODE REVIEW

File: src/collapse/vtune.rs [hunk 1]
```diff
@@ -1,10 +1,19 @@
 use std::io::{self, BufRead};
 
-use log::{error, warn};
+use log::warn;
 
 use crate::collapse::common::Occurrences;
 use crate::collapse::Collapse;
 
+macro_rules! invalid_data {
```
Comment (jonhoo) [inline src/collapse/vtune.rs]: We can probably share this macro across many collapsers.

File: tests/common/collapse.rs [hunk 2]
```diff
@@ -104,3 +104,27 @@ where
     collapser.collapse(r, std::io::sink()).unwrap();
     test_logger::validate(asserter);
 }
+
+pub fn test_collapse_error<C>(
```
Comment (jonhoo) [inline tests/common/collapse.rs]: Is this useful for other collapsers too?

File: src/collapse/common.rs [hunk 3]
```diff
@@ -11,6 +11,15 @@ use crossbeam::channel;
 use fnv::FnvHashMap;
 use lazy_static::lazy_static;
 
+macro_rules! invalid_data_error {
```
Comment (jonhoo) [inline src/collapse/common.rs]: Are there other collapsers that could use this right now?

File: tests/common/collapse.rs [hunk 4]
```diff
@@ -104,3 +104,24 @@ where
     collapser.collapse(r, std::io::sink()).unwrap();
     test_logger::validate(asserter);
 }
+
+pub fn test_collapse_error<C>(mut collapser: C, test_filename: &str) -> io::Error
```
Comment (jonhoo) [inline tests/common/collapse.rs]: Are there other collapsers that could use this right now?

File: src/collapse/common.rs (hunk 3 above)
Comment (jasonrhansen) [reply src/collapse/common.rs]: Yes, I'll do a separate PR for changes to the sample collapser that will use this.

File: tests/common/collapse.rs (hunk 4 above)
Comment (jasonrhansen) [reply tests/common/collapse.rs]: Same here, I'll use it for the sample collapser.

**Pull request merged**